include/engine/api/trip_parameters.hpp

```diff
@@ -47,7 +47,27 @@ namespace api
  */
 struct TripParameters : public RouteParameters
 {
-    // bool IsValid() const; Falls back to base class
+    TripParameters()
+        : RouteParameters(false,
+                          false,
+                          false,
+                          RouteParameters::GeometriesType::Polyline,
+                          RouteParameters::OverviewType::Simplified,
+                          {}),
+          source(-1), destination(-1)
+    {
+    }
+
+    template <typename... Args>
+    TripParameters(int source_, int destination_, Args &&... args_)
+        : RouteParameters{std::forward<Args>(args_)...}, source{source_}, destination{destination_}
+    {
+    }
+
+    int source;
+    int destination;
```

Comment:
Source and destination are indices - they should really be `std::size_t` and maybe even wrapper in an `optional<T>`